src/solvers/mp/MasterProblem.cpp

```diff
@@ -0,0 +1,1278 @@
+			if(sk!=0) rep << "\t";
+
+			char buffer0[20];
+			string skill = pScenario_->intToSkill_[sk];
+			sprintf(buffer0, "%-12s", skill.c_str());
+			rep << buffer0;
+
+			for (int day = firstDay; day < firstDay+nbDays; day++){
+				double shiftValue = pModel_->getVarValue(skillsAllocVars_[day][s-1][sk]);
+				char buffer[100];
+				if(abs(shiftValue - round(shiftValue)) < EPSILON)
+					sprintf(buffer, "|%4d", (int) round(shiftValue));
+				else sprintf(buffer, "|%2.2f", shiftValue);
+				rep << buffer;
+				if(Tools::isSunday(day)) rep << "| ";
+			}
+
+			rep << "|" << std::endl;
+		}
+	}
+	rep << std::endl;
+
+	cout << rep.str();
+
+	return rep.str();
+}
+
+string MasterProblem::workedWeekendsToString(bool printInteger){
+	return "";
+}
+
+//---------------------------------------------------------------------------
+//
+// STAB: Methods required to implement stabilization in the column generation
+//
+//---------------------------------------------------------------------------
+
+// STAB
+// Multiply the upper bound of the input variable by the input factor
+void MasterProblem::multiplyUbInSolver(MyVar* pVar, OsiSolverInterface* solver, double factor) {
+	int varind = pVar->getIndex();
+	double ub = pVar->getUB();
+
+	if (ub != solver->getColUpper()[varind]) {
+		Tools::throwError("multiplyUbInSolver: the upper bound stored in the variable is not the same as that in the solver!");
+	}
+
+	solver->setColUpper(varind,factor*ub);
+	pVar->setUB(factor*ub);
+}
+
+// STAB
+// Set the bound of the input variable to the input value
+void MasterProblem::updateVarUbInSolver(MyVar* pVar, OsiSolverInterface* solver, double value) {
+	int varind = pVar->getIndex();
+	double ub = pVar->getUB();
+
+	if (ub != solver->getColUpper()[varind]) {
+		Tools::throwError("updateVarUbInSolver: the upper bound stored in the variable is not the same as that in the solver!");
+	}
+
+	solver->setColUpper(varind,value);
+	pVar->setUB(value);
+}
+
+// STAB
+// Set the cost of the input variable to the input value
+void MasterProblem::updateVarCostInSolver(MyVar* pVar, OsiSolverInterface* solver, double value) {
+	int varind = pVar->getIndex();
+	double cost = pVar->getCost();
+
+	if (cost != solver->getObjCoefficients()[varind]) {
+		Tools::throwError("updateVarCostInSolver: the cost stored in the variable is not the same as that in the solver!");
+	}
+
+	solver->setObjCoeff(varind,value);
+	pVar->setCost(value);
+}
+
+// STAB
+// Update all the upper bounds of the stabilization variables by multiplying
+// them by an input factor
+void MasterProblem::stabUpdateBound(OsiSolverInterface* solver, double factor) {
+	for(int i=0; i<pScenario_->nbNurses_; i++){
+		// stabilization variables corresponding to the global constraints of
+		// of the nurses
+		multiplyUbInSolver(stabMinWorkedDaysPlus_[i], solver, factor);
+		multiplyUbInSolver(stabMaxWorkedDaysMinus_[i], solver, factor);
+		multiplyUbInSolver(stabMaxWorkedWeekendMinus_[i], solver, factor);
+	}
```

Comment:
True in every method related to stabilization: the minimum and maximum number of working days are in the MP only in rotation decomposition. These lines should be moved to subclass I guess.

Reply:
Moved in next major refactoring.